src/Spout/Writer/XLSX/Manager/Style/StyleRegistry.php

```diff
@@ -48,11 +121,58 @@ public function registerStyle(Style $style)
     {
         $registeredStyle = parent::registerStyle($style);
         $this->registerFill($registeredStyle);
+        $this->registerFormat($registeredStyle);
         $this->registerBorder($registeredStyle);
 
         return $registeredStyle;
     }
 
+    /**
+     * Register a format definition
+     *
+     * @param Style $style
+     */
+    protected function registerFormat(Style $style)
+    {
+        $styleId = $style->getId();
+
+        $format = $style->getFormat();
+        if ($format) {
+            $isFormatRegistered = isset($this->registeredFormats[$format]);
+
+            // We need to track the already registered format definitions
+            if ($isFormatRegistered) {
+                $registeredStyleId = $this->registeredFormats[$format];
+                $registeredFormatId = $this->styleIdToFormatsMappingTable[$registeredStyleId];
+                $this->styleIdToFormatsMappingTable[$styleId] = $registeredFormatId;
+            } else {
+                $this->registeredFormats[$format] = $styleId;
+
+                if (isset(self::$builtinNumFormatToIdMapping[$format])) {
```

Comment:
nitpick: can be simplified `$id = self::$builtinNumFormatToIdMapping[$format] ?? $this->formatIndex++;`